query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -96,6 +95,18 @@ private String functionToStringForAvgRate(
         columnName, (double) aggregateIntervalInSeconds / rateIntervalInSeconds);
   }
 
+  private Optional<Expression> getColumnNameForAvgRate(Function function) {
+    return function.getArgumentsList().stream().filter(e -> e.hasColumnIdentifier()).findFirst();
+  }
+
+  private String getRateIntervalForAvgRate(Function function) {
```

Comment:
We have integration-tests in query-service. We should definitely add one for this case.

Reply:
We already have two integration tests for testing the average rate changes. Should I add for this specific case also where average rate is in `order-by` expression ?

Reply:
Yes. We should definitely add one.

Reply:
resolved